File: ui/palmai.js
const context = "only respond with one liners";
const examples = [];

// let messages = [
//     "please respond with one liners",
//     "Sure, I will respond with one liners.",
//     "Can you interview me on react.js?",
//     "Sure, first question, what is useState?",
//     "useState is a react hook",
//     "Correct, next question, what is useEffect?",
//     "useEffect is a react hook",
// ];

let messages = [
    "respond with a maximum of 10 words ONLY",
    "Sure, I will respond with a MAXIMUM of 10 words.",
    "Can you interview me on react.js?",
    // "Sure, first question, what is useState?",
    // "useState is a react hook",
    // "Correct, next question, what is useEffect?",
    // "useEffect is a react hook",
];



export default async function think ( user_message=null )
{
    console.log( "thinking..." );
    console.log( "" );

    
    let response = await fetch( "http://127.0.0.1:5000/think", {
        method: "POST",
        headers: {
            "Content-Type": "application/json"
        },
        body: JSON.stringify( {
            // "context": context,
            // "examples": examples,
            "messages": messages
        } )
    } );
    
    if ( user_message !== null )
    {
        messages.push( user_message );
    }
    // update message withr response
    messages = await response.json();
    return messages[ messages.length - 1 ];
}

// think( "howdy neighbour!" ).then( console.log );

File: AI Interviewv2/api/palmai.py
import os
import google.generativeai as palm

palm.configure(api_key="AIzaSyBri5O5YMK_qCfKxvPl2CjbfBQ6iRJYLh4")

context = """
You will be an interviewer. You will ask the user questions based on reactjs. you will ask one question at a time. then you will tell the user whether the answer is 'Correct' or 'Wrong'. You will not say anything else other than correct or wrong. then you will ask the next question. Answer with 10 words maximum.
"""
# """
# At the command line, only need to run once to install the package via pip:
# $ pip install google-generativeai
# """


defaults = {
    "model": "models/chat-bison-001",
    "temperature": 0,
    "candidate_count": 1,
    "top_k": 4,
    "top_p": 0.95,
}
# examples = []
# messages = ["please keep your responses extremely concise, one line maximum", "Sure, I will respond with one liners."]


def think(context=context, examples=[], messages=[]):
    # context = """
    # * Interview the user about their name, age, occupation, and hobbies.
    # * Ask the user about their favorite movie, book, and song.
    # * Ask the user about their dream job and what they want to achieve in life.
    # * Ask the user about their thoughts on the future and what they hope to see happen in the world.
    # * Thank the user for their time and participation in the interview.
    # """
    response = palm.chat(
        **defaults, context=context, examples=examples, messages=messages
    )

    ai_message = response.last
    messages.append(ai_message)

    return messages


if __name__ == "__main__":
    while True:
        user_message = input()
        print(think(user_message))
